worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionDataWriter.java

```diff
@@ -98,34 +99,41 @@ public FileWriter(
       PartitionType partitionType,
       boolean rangeReadFilter)
       throws IOException {
-    this.fileInfo = fileInfo;
-    this.flusher = flusher;
-    this.flushWorkerIndex = flusher.getWorkerIndex();
+    this.storageManager = storageManager;
+    CreateFileResult createFileResult = storageManager.createFile(createFileContext, this);
```

Comment:
I think it's not a good idea to call `createFile` in constructor here, and I think it's unnecessary to define `CreateFileResult`. Instead, should call `createFile` in `StorageManager#createPartitionDataWriter` and pass `flusher` and `fileInfo` as parameters to the constructor.